pkg/sync/adapter.go

```diff
@@ -210,6 +245,34 @@ func (a Adapter) pushLocal(path string, files []string, delFiles []string, isFor
 	return nil
 }
 
+// isRemoteFolderEmpty returns true if the 'syncFolder' path in the pod contains no files/folders, false otherwise
+func isRemoteFolderEmpty(a Adapter, compInfo common.ComponentInfo, syncFolder string) (bool, error) {
+
+	stdoutWriter, stdoutOutputChan := exec.CreateConsoleOutputWriterAndChannel()
+	stderrWriter, stderrOutputChan := exec.CreateConsoleOutputWriterAndChannel()
+
+	err := exec.ExecuteCommand(a.Client, compInfo, []string{"ls", "-a", syncFolder}, false, stdoutWriter, stderrWriter)
+	if err != nil {
```

Comment:
Can be done in a more talkative way and more importantly we can make err scope at block level instead of the function level. For example

```
if err := exec.ExecuteCommand(
		a.Client,
		compInfo,
		[]string{"ls", "-a", syncFolder},
		false,
		stdoutWriter,
		stderrWriter,
); err != nil {
		return false, err
}
```

Reply:
Done.